subprojects/test-runner/test-report/src/main/kotlin/com/avito/android/test/report/ReportState.kt

```diff
@@ -40,8 +46,8 @@ sealed class ReportState {
                 var dataSet: DataSet? = null,
                 var startTime: Long,
                 var endTime: Long = 0,
-                var preconditionStepList: MutableList<StepResult> = mutableListOf(),
-                var testCaseStepList: MutableList<StepResult> = mutableListOf()
+                var preconditionStepList: CopyOnWriteArrayList<StepResult> = CopyOnWriteArrayList(),
```

Comment:
Do we need to know about CopyOnWriteArrayList in constructor signature?

Reply:
Yes, because users could pass a wrong `MutableList` without that restriction

Reply:
Do we need to make the same in `StepAttachments`?

Reply:
StepAttachments doesn't allow to pass collections through constructor, so no